src/irmin-pack/s.ml

```diff
@@ -79,6 +79,12 @@ module type S = sig
     ([> `Msg of string ], [> `Msg of string ]) result Lwt.t
 
   val reconstruct_index : ?output:string -> Irmin.config -> unit
+
+  val traversal :
+    ?print:(string -> unit) ->
```

Comment:
Let's not bother with validating the flattened path in this PR. Doing this will be a one time job that don't necessitate support in irmin/master. If you fully drop the support for it, it will make your code much cleaner.